cpp/src/parquet/stream_reader.cc

```diff
@@ -173,6 +174,60 @@ StreamReader& StreamReader::operator>>(std::string& v) {
   return *this;
 }
 
+StreamReader& StreamReader::operator>>(::arrow::Decimal128& v) {
```

Comment:
Is there anyway to avoid repeating the implementation here?  For example, would something like this work?

```
StreamReader& StreamReader::operator>>(::arrow::Decimal128& v) {
  std::optional<::arrow::Decimal128> maybe_v;
  this >> maybe_v;
  if (!maybe_v) {
    // I'm not sure what happens here, maybe throw an exception?
  }
  v = *maybe_v;
  return this;
}
```

Reply:
Tried to reuse code and throw if it is null.